nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -129,16 +129,46 @@ public static TreePath findEnclosingMethodOrLambdaOrInitializer(TreePath path) {
   }
 
   /**
-   * @param element the element
-   * @return all annotations on the element and on the type of the element
+   * NOTE: this method does not work for getting all annotations of parameters of methods from class
+   * files. For that case, use {@link #getAllAnnotationsForParameter(Symbol.MethodSymbol, int)}
+   *
+   * @param symbol the symbol
+   * @return all annotations on the symbol and on the type of the symbol
    */
-  public static Iterable<? extends AnnotationMirror> getAllAnnotations(Element element) {
+  public static Stream<? extends AnnotationMirror> getAllAnnotations(Symbol symbol) {
     // for methods, we care about annotations on the return type, not on the method type itself
-    TypeMirror typeMirror =
-        element instanceof Symbol.MethodSymbol
-            ? ((Symbol.MethodSymbol) element).getReturnType()
-            : element.asType();
-    return Iterables.concat(element.getAnnotationMirrors(), typeMirror.getAnnotationMirrors());
+    Stream<? extends AnnotationMirror> typeUseAnnotations = getTypeUseAnnotations(symbol);
+    return Stream.concat(symbol.getAnnotationMirrors().stream(), typeUseAnnotations);
+  }
+
+  /**
+   * Works for method parameters defined either in source or in class files
+   *
+   * @param symbol the method symbol
+   * @param paramInd index of the parameter
+   * @return all declaration and type-use annotations for the parameter
+   */
+  public static Stream<? extends AnnotationMirror> getAllAnnotationsForParameter(
+      Symbol.MethodSymbol symbol, int paramInd) {
+    Symbol.VarSymbol varSymbol = symbol.getParameters().get(paramInd);
+    return Stream.concat(
+        varSymbol.getAnnotationMirrors().stream(),
+        symbol
+            .getRawTypeAttributes()
+            .stream()
+            .filter(
+                t ->
+                    t.position.type.equals(TargetType.METHOD_FORMAL_PARAMETER)
+                        && t.position.parameter_index == paramInd));
```

Comment:
Hrm, I think this will only return type use annotations.  If you look at the [`TargetType` code](https://github.com/google/error-prone-javac/blob/a68a002c508b9b862ab4b966c70aad0a1686d328/src/jdk.compiler/share/classes/com/sun/tools/javac/code/TargetType.java#L45) it seems to only be talking about what type of node an annotation is targeting.  So I *think* anything with a `TargetType` should be an annotation.  Do you have a suggested test to check this?

Reply:
Spoke offline, and we're going to go ahead under the assumption that `getRawTypeAttributes()` is behaving as one would expect